api/src/OpenTelemetry/Trace/Core.hs

```diff
@@ -688,10 +690,12 @@ spanIsRemote (Dropped _) = pure False
  to semantic versioning .
 -}
 unsafeReadSpan :: (MonadIO m) => Span -> m ImmutableSpan
-unsafeReadSpan = \case
-  Span ref -> liftIO $ readIORef ref
-  FrozenSpan _s -> error "This span is from another process"
-  Dropped _s -> error "This span was dropped"
+unsafeReadSpan s =
+  toImmutableSpan s >>= \case
+    Right span -> pure span
+    Left frozenOrDropped -> case frozenOrDropped of
+      SpanFrozen -> error "This span is from another process"
+      SpanDropped -> error "This span was dropped"
```

Comment:
just outta curiosity, why the `error` call vs throwing an exception?

is the intention to preserve the thunk so that a frozen/dropped span doesn't immediately halt a test unless it's inspected?

Reply:
Honestly, it's just that the code here has existed since the beginning and never been an issue. It would probably make sense to throw here, but it's a relatively niche function that has t needed much changing.